### ISSUE TITLE
TransportSerializationException during rapse SearchResponse

### ISSUE THREAD

**Issue by hleb-albau** [strapdata/elassandra#141]
Java build 1.8.0_151-b12, Java 9.01, Ellassandra 5.5.0.7(6), elastic driver 5.5.3(0,1,2)
I do next query(generally speaking all my search queries):

```
        val elasticResponse = elasticClient.prepareSearch("test-index") .get()
```

If my index is empty, than i successfully obtain SearchResponse object. If any number of items present in index, than i have next error:
```
 TransportSerializationException[Failed to deserialize response of type [org.elasticsearch.action.search.SearchResponse]]; nested: IndexOutOfBoundsException;
	at org.elasticsearch.transport.TcpTransport.handleResponse(TcpTransport.java:1431)
	at org.elasticsearch.transport.TcpTransport.messageReceived(TcpTransport.java:1403)
	at org.elasticsearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:74)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:310)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:297)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:413)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:265)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1334)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:926)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:134)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:644)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:544)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:498)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458)
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:858)
	at java.base/java.lang.Thread.run(Thread.java:844)
Caused by: java.lang.IndexOutOfBoundsException
	at org.elasticsearch.transport.netty4.ByteBufStreamInput.readBytes(ByteBufStreamInput.java:142)
	at org.elasticsearch.common.io.stream.FilterStreamInput.readBytes(FilterStreamInput.java:45)
	at org.elasticsearch.common.io.stream.StreamInput.readBytesReference(StreamInput.java:143)
	at org.elasticsearch.common.io.stream.StreamInput.readOptionalText(StreamInput.java:304)
	at org.elasticsearch.search.SearchHit.readFrom(SearchHit.java:847)
	at org.elasticsearch.search.SearchHit.readSearchHit(SearchHit.java:840)
	at org.elasticsearch.search.SearchHits.readFrom(SearchHits.java:216)
	at org.elasticsearch.search.SearchHits.readSearchHits(SearchHits.java:202)
	at org.elasticsearch.search.internal.InternalSearchResponse.<init>(InternalSearchResponse.java:53)
	at org.elasticsearch.action.search.SearchResponse.readFrom(SearchResponse.java:306)
	at org.elasticsearch.transport.TcpTransport.handleResponse(TcpTransport.java:1428)
```

Also i get some debug here and find next points:
* if i have one item in index -> search hit parsed correctly and than it fails at parsing InternalSearchResponse items
* if i have n items -> first search hit parsed correctly, and that second one fails.

**Comment by vroyer** [comment]
Ok, there is a transport issue with the native elasticsearch transport client. This is fixed on the master branch, and will be in the next release tomorrow.

**Comment by vroyer** [comment]
Fixed in release 5.5.08 by this [commit](https://github.com/strapdata/elassandra/commit/894171ace201255e57e94f821c27cc46f0acc146)
Thanks for reporting the issue.

**Comment by hleb-albau** [comment]
Thanks!
